### ISSUE TITLE
Wrong datetime format

### ISSUE THREAD

**Issue by lucidfrontier45** [rethinkdb/logstash-input-rethinkdb#4]
This input plugin currently gives off RethinkDB time value type as a string like "2015-12-25 06:16:15 +0000".
The format should be like "2015-12-25T06:16:15Z+0000" so that Elasticsearch can easily parse it as datetime type.

**Comment by deontologician** [comment]
Ah, thanks. I'll fix this today

**Comment by deontologician** [comment]
OK, after looking into this a bit, I'm thinking the fix for this should be for the driver to send on raw pseudotypes rather than just format dates differently. Right now, they're turned into native date Ruby objects and the logstash machinery serializes them, probably using the `.to_str` representation.

Since the logstash input can be used with many other outputs (like SOLR, etc), it wouldn't make sense to change the format to something just for elasticsearch. In addition, it would leave other pseudotypes unconverted.

So the fix is going to be outputting the raw pseudotype form, which doesn't lose information. For a date it will look like:

``` json
{"my_date": {
  "$reql_type$": "TIME",
  "epoch_time": 1449274123,
  "timezone": "+00:00"
}}
```

Then on the elasticsearch end, or in your logstash filter/configuration, you can extract the piece that's best for your use case.

Elasticsearch can adjust the formats it accepts: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-date-format.html#built-in-date-formats

**Comment by lucidfrontier45** [comment]
BTW, I rechecked the output by rubydebug codec and the output of this input plugin was actually _not_ string.

The output was like this. 

```
{
  "new_val" => {
    "msg" => {   
      "aa" => 1,   
      "bb" => 2    
    },
    "timestamp" => 2015-12-25 15:16:15 +0900,
    "user_id" => "test"
  },
  "@version" => "1",
  "@timestamp" => "2015-12-29T01:18:51.315Z",
  "@metadata" => {
    "index" => "data",
    "document_type" => "app_logs",
    "document_id" => "e31c462e-b051-44f5-8c78-710bcafc6a31"
  }
}

```

`@timestamp` is what logstash automatically append and looks like a string since it is quoted.
On the other hand `new_val.timestamp` comes from rethinkdb without being quoted. 

To what data type will rethinkdb date be interpreted?

**Comment by deontologician** [comment]
The driver turns the time pseudotype into `Time` objects in Ruby. That date object may be present in debugging, but it ultimately gets serialized when sent over logstash

**Comment by deontologician** [comment]
OK, I've released version 0.2.3 on ruby gems @lucidfrontier45 let me know if this works out for you.

**Comment by lucidfrontier45** [comment]
@deontologician 

It works. Thank you.
